### PR TITLE
Added footer and case studies page

### PR DESCRIPTION
**Pull request by DevTaehong**
- Added footer and case studies page
- Restructured SVG components
- Restructured folders

Screenshot 2023-09-14 at 3 57 19 PM
Screenshot 2023-09-14 at 3 56 59 PM

### CODE REVIEW

File: components/CaseStudiesHero.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            <span key={text} className="relative inline-block">
+              <span
+                className={`${
+                  text === "Case" && "sm:w-[15rem]"
+                } absolute top-[1.88rem] h-[0.8125rem] w-full shrink-0 bg-accentSecondary 
+                lg:top-[3.6rem] lg:h-[1.68963rem]`}
+              />
+              <span className="relative">{text}</span>
+            </span>{" "}
+            {text !== "Studies" && <br className="block sm:hidden" />}
```
Comment (Nassim-Saboundji) [inline components/CaseStudiesHero.tsx:19]: You shouldn't use the br element for spacing. You should instead use padding or margins to create the spacing you need.

**Review by Nassim-Saboundji — CHANGES_REQUESTED** [review]
Good job!

I noticed that there is a lot of code to handle svg states (onHover, etc...) that is repeated from one svg to the next. Can you look into how you could refactor your code so you don't have to this much repetition.

**Comment by cdsalty** [comment]
@DevTaehong You are still not rendering any text when the user is on 'light mode.' Could you be sure to correct this as well?
Screen Shot 2023-09-14 at 7 19 44 PM

**Comment by DevTaehong** [comment]
> @DevTaehong You are still not rendering any text when the user is on 'light mode.' Could you be sure to correct this as well? Screen Shot 2023-09-14 at 7 19 44 PM

My bad :( forgot to change one variable. I've made sure it works.

**Comment by DevTaehong** [comment]
@Nassim-Saboundji @cdsalty I've made the requested changes. Thanks for the feedback. the code looks much better.

File: components/navBar/NavBarMenuButton.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const NavBarMenuButton = () => {
+function NavBarMenuButton() {
   return (
     <svg
```
Comment (cdsalty) [inline components/navBar/NavBarMenuButton.tsx:3]: Place this inside an SVG and import only the SVG if possible.

File: components/CaseStudiesHero.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const heroText = ["Case", "Studies"];
```
Comment (cdsalty) [inline components/CaseStudiesHero.tsx:1]: Move this inside its own index file along with any other dummy data or hard-coded data.

File: components/CaseStudiesHero.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            <span key={text} className="relative inline-block">
+              <span
+                className={`${
+                  text === "Case" && "sm:w-[15rem]"
+                } absolute top-[1.88rem] h-[0.8125rem] w-full shrink-0 bg-accentSecondary 
+                lg:top-[3.6rem] lg:h-[1.68963rem]`}
+              />
+              <span className="relative">{text}</span>
+            </span>{" "}
+          </>
```
Comment (cdsalty) [inline components/CaseStudiesHero.tsx:19]: Here are a few things to review:
- Consider making this a reusable component.
- You don't need the fragment, A `<div>` would be more appropriate here. Also, you would have placed your key in the fragment, not the `<span>`
- I understand you don't have much data but it's still good practice to use a unique key when possible. 

```
// projectsData.ts
export const projects = [
  // ...
];

```

```
// Import it in your component
import { projects } from './projectsData';

```

File: components/CaseStudiesProject.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import { Link } from "@nextui-org/link";
+import Image from "next/image";
+
+const CaseStudiesProject = ({
+  project,
+}: {
+  project: { name: string; description: string; image: string; color: string };
+}) => {
+  return (
+    <>
```
Comment (cdsalty) [inline components/CaseStudiesProject.tsx:11]: I don't think the React Fragment is necessary here.

File: components/CaseStudiesProject.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      <Card
+        as={Link}
+        href={`/case-studies/${project.name.toLowerCase()}`}
+        className={`${project.color} items-center rounded-[0.62725rem] px-[1.54rem] pt-[2.73rem] xl:rounded-2xl xl:px-[3.75rem] xl:pt-[4.35rem]`}
+      >
+        <Image
+          className="lg:h-[15.8rem] lg:w-[26.8rem]"
+          src={project.image}
+          width={278.486}
+          height={165.054}
```
Comment (cdsalty) [inline components/CaseStudiesProject.tsx:21]: FYI: The width and height attributes are used to infer the correct aspect ratio of the image and avoid layout shift from the image loading in. The width and height do not determine the rendered size of the image file. Learn more about [Image Sizing](https://nextjs.org/docs/app/building-your-application/optimizing/images#image-sizing).
You can also read more about it here: https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images

File: components/CaseStudiesProjects.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      >
+        {projects.map((project) => (
+          <div
+            key={project.name}
+            className="max-w-[21.5625rem] lg:max-w-[34.375rem]"
+          >
+            <CaseStudiesProject project={project} />
+          </div>
+        ))}
+      </div>
```
Comment (cdsalty) [inline components/CaseStudiesProjects.tsx:51]: As mentioned earlier, you could move these into a separate component.

File: components/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+const links = [
+  {
+    link: "https://www.linkedin.com/in/taehong/",
+    icon: LinkedInIcon,
+  },
+  {
+    link: "https://github.com/DevTaehong",
+    icon: FooterGitHubIcon,
+  },
+];
```
Comment (cdsalty) [inline components/Footer.tsx:19]: Move this data outside the component as well.

**Review by cdsalty — CHANGES_REQUESTED** [review]
Looking great @DevTaehong! To further align it with our project specifications, please consider the following updates:

Morrent Project Background: Update the background to the blue color specified in the design to enhance section differentiation.

Text Color in Light Mode: The text on the Morrent project card is currently set to white in light mode. Please adjust this for better readability.

Footer Alignment: The footer is well-designed but should be aligned at the bottom of the div as per the design guidelines.

Tablet Responsiveness: Verify the responsiveness of the 'Work Experience' section, particularly when viewed on tablets.

Error Codes: Several errors have been logged in your developer console. Please make sure that these are resolved before your next submission.

Extract SVG to a Separate File
Consider storing the SVG asset in a separate SVG file if it is static and doesn't change. This makes the React component easier to read and manage. You can then import and use it in your React component.


Please review the code suggestions I've provided in your pull request. You're doing excellent work, and by focusing on reusable components and externalizing hardcoded data, we can elevate the project even further.


Could you make sure that the PR contains only the changes mentioned above?

Thank you for your attention to these details, and keep up the great work!

File: components/navBar/NavBarMenuButton.tsx (hunk 2 above)
Comment (DevTaehong) [reply components/navBar/NavBarMenuButton.tsx:3]: Screenshot 2023-09-15 at 2 37 25 AM

I'm using it as an icon for NexUI and the icon prop should be ReactNode.

File: components/CaseStudiesProject.tsx (hunk 5 above)
Comment (DevTaehong) [reply components/CaseStudiesProject.tsx:11]: Screenshot 2023-09-15 at 2 55 03 AM

Without the Fragment, it causes an error. Did you mean to use a div tag instead?

Screenshot 2023-09-15 at 2 56 25 AM

**Comment by DevTaehong** [comment]
@cdsalty 
Morrent Project Background: I've made sure that it renders the colours properly according to the design.

Text Color in Light Mode: The blue color is rendering properly so I've applied the text color that the design has.

Footer Alignment: Now it's aligned. Thanks for mentioning it.

Tablet Responsiveness: I've changed it.

Error Codes: To get rid of all errors I had, I've changed all components related to the errors.

Extract SVG to a Separate File: I've only made svg assets React components when the colour needs to be changed in dark mode. Is there another way I can change SVG's colour without making components?

Thank you for your feedback! I was missing a lot of things lol

**Review by Nassim-Saboundji — APPROVED** [review]
Looks good!

When it comes to SVGs. I usually recommended using [SVGR](https://react-svgr.com/) which allows you to import svgs as React components that are optimized. Now for modifying the colors you need to jumps through some hoops but at least you don't need to create multiple svgs of different colors.

I recommend reading [this](https://discord.com/channels/958989850181259274/1152308222036807823) thread on discord that explains how to change an SVGs color with SVGR.

You can make further changes related to SVGs in a future PR.

**Comment by DevTaehong** [comment]
> Looks good!
> 
> When it comes to SVGs. I usually recommended using [SVGR](https://react-svgr.com/) which allows you to import svgs as React components that are optimized. Now for modifying the colors you need to jumps through some hoops but at least you don't need to create multiple svgs of different colors.
> 
> I recommend reading [this](https://discord.com/channels/958989850181259274/1152308222036807823) thread on discord that explains how to change an SVGs color with SVGR.
> 
> You can make further changes related to SVGs in a future PR.

Thanks! will look into it!

**Pull request merged**